server/src/main/java/org/apache/iotdb/db/engine/flush/FlushManager.java

```diff
@@ -136,10 +136,8 @@ public void registerTsFileProcessor(TsFileProcessor tsFileProcessor) {
       if (LOGGER.isDebugEnabled()) {
         if (tsFileProcessor.isManagedByFlushManager()) {
           LOGGER.debug(
-              "{} is already in the flushPool, the first one: {}, the given processor flushMemtable number = {}",
+              "{} is already in the flushPool, the given processor flushMemtable number = {}",
               tsFileProcessor.getTsFileResource().getTsFile().getAbsolutePath(),
-              tsFileProcessorQueue.isEmpty() ? "empty now"
-                  : tsFileProcessorQueue.getFirst().getStorageGroupName(),
               tsFileProcessor.getFlushingMemTableSize());
```

Comment:
Thanks for bringing this up. Relook this process may have other problems...

Current thread setManagedByFlushManager(true).
FlushThread flush a memtable and setManagedByFlushManager(false)
Current thread check tsfileProcessor.isManagedByFlushManager() return false

So, it's better to put all logs before submit(FlushThread)   line 126.

Reply:
before line 126 or line 125?

Reply:
Thank you all, I've reorganized the debug logs. PTAL